server/src/main/java/org/elasticsearch/index/engine/ThreadPoolMergeExecutorService.java

```diff
@@ -252,6 +240,33 @@ private static long newTargetIORateBytesPerSec(
         return newTargetIORateBytesPerSec;
     }
 
+    static class AtomicIORate extends AtomicLong {
```

Comment:
I'd prefer to use delegation here, i.e., to let `AtomicIORate` have a `final AtomicLong rate` field instead. This way it is obvious that only exposed methods here are called. Whereas otherwise callers could call any `AtomicLong` method too.

Reply:
Right! I thought `AtomicLong` did not expose the `weakCompareAndSetVolatile` method that's used in the `updateAndGet` implementation, but it does. Pushed https://github.com/elastic/elasticsearch/pull/120869/commits/79030ed51d0251582ccf83487b9b0d314f8d5cf0